pkg/proxy/kafka.go

```diff
@@ -159,10 +178,192 @@ func (k *kafkaRedirect) canAccess(req *kafka.RequestMessage, numIdentity policy.
+		if ip != nil && ip.To4() == nil {
+			version = accesslog.VersionIPV6
+		}
+	}
+
+	// At egress, the local origin endpoint is the source
+	if !r.ingress {
+		r.localEndpointInfo(&info)
+	} else if err == nil {
+		if srcIdentity != 0 {
+			r.getInfoFromConsumable(ipstr, &info, srcIdentity)
+		} else {
+			// source security identity 0 is possible when somebody else
+			// other than the BPF datapath attempts to
+			// connect to the proxy.
+			// We should log no source information in that case, in the proxy log.
+			log.Warn("Missing security identity in source endpoint info")
+		}
+
+	}
+
+	return info, version
+}
+
+// fillReservedIdentity resolves the labels of the specified identity if known
+// locally and fills in the following info member fields:
+//  - info.Identity
+//  - info.Labels
+//  - info.LabelsSHA256
+func (r *kafkaRedirect) fillReservedIdentity(info *accesslog.EndpointInfo, id policy.NumericIdentity) {
+	info.Identity = uint64(id)
+
+	if c := policy.GetConsumableCache().Lookup(id); c != nil {
+		if c.Labels != nil {
+			info.Labels = c.Labels.Labels.GetModel()
+			info.LabelsSHA256 = c.Labels.GetLabelsSHA256()
+		}
+	}
+}
+
+// egressDestinationInfo returns the destination EndpointInfo for a flow
+// leaving the proxy at egress.
+func (r *kafkaRedirect) egressDestinationInfo(ipstr string, info *accesslog.EndpointInfo) {
+	ip := net.ParseIP(ipstr)
+	if ip != nil {
+		if ip.To4() != nil {
+			info.IPv4 = ip.String()
+
+			if nodeaddress.IsHostIPv4(ip) {
+				r.fillReservedIdentity(info, policy.ReservedIdentityHost)
+				return
+			}
+
+			if nodeaddress.GetIPv4ClusterRange().Contains(ip) {
+				c := addressing.DeriveCiliumIPv4(ip)
+				ep := endpointmanager.LookupIPv4(c.String())
+				if ep != nil {
+					info.ID = uint64(ep.ID)
+					info.Labels = ep.GetLabels()
+					info.LabelsSHA256 = ep.GetLabelsSHA()
+					info.Identity = uint64(ep.GetIdentity())
+				} else {
+					r.fillReservedIdentity(info, policy.ReservedIdentityCluster)
+				}
+			} else {
+				r.fillReservedIdentity(info, policy.ReservedIdentityWorld)
+			}
+		} else {
+			info.IPv6 = ip.String()
+
+			if nodeaddress.IsHostIPv6(ip) {
+				r.fillReservedIdentity(info, policy.ReservedIdentityHost)
+				return
+			}
+
+			if nodeaddress.GetIPv6ClusterRange().Contains(ip) {
+				c := addressing.DeriveCiliumIPv6(ip)
+				id := c.EndpointID()
+				info.ID = uint64(id)
+
+				ep := endpointmanager.LookupCiliumID(id)
+				if ep != nil {
+					ep.RLock()
+					info.Labels = ep.GetLabels()
+					info.LabelsSHA256 = ep.GetLabelsSHA()
+					info.Identity = uint64(ep.GetIdentity())
+					ep.RUnlock()
+				} else {
+					r.fillReservedIdentity(info, policy.ReservedIdentityCluster)
+				}
+			} else {
+				r.fillReservedIdentity(info, policy.ReservedIdentityWorld)
+			}
+		}
+	}
+}
+
+func (r *kafkaRedirect) getDestinationInfo(dstIPPort string) accesslog.EndpointInfo {
+	info := accesslog.EndpointInfo{}
+	ipstr, port, err := net.SplitHostPort(dstIPPort)
+	if err == nil {
```

Comment:
If `dstIPPort` returns an error, why continue the execution of this function?

Add debug log to show the error in case we ever it this.